CMakeLists.txt

```diff
@@ -278,10 +306,11 @@ set_target_properties(
     PROPERTIES
     CMAKE_CXX_STANDARD 17
     CMAKE_CXX_STANDARD_REQUIRED True
-    LINKER_LANGUAGE CXX 
+    LINKER_LANGUAGE CXX
+    LINK_FLAGS_RELEASE "-flto=auto -s"
```

Comment:
Once again, I'm not sure if these are compiler specific, but if they are could we wrap this with an `if(COMPILER)` check as well?